### PR TITLE
Add get_relative_path() to base filesystem methods

### PR DESCRIPTION
**Pull request by gui-marc**
### Feature Description
This PR introduces the get_relative_path function, which computes the relative path from a given base path to a specified file. The function signature is as follows:

```cpp
std::string get_relative_path(const std::string& filename, const std::string& base_path);
```

### Functionality

The get_relative_path function takes two parameters:

- filename: The absolute path of the target file.
- base_path: The absolute path of the base directory.

The function returns a string representing the relative path from base_path to filename. This is useful in scenarios where relative paths are needed for portability, readability, or other path manipulations.

### Usage Examples

Below are some examples demonstrating how to use the get_relative_path function:

```cpp
std::string filename = "/home/user/projects/my_project/src/main.cpp";
std::string base_path = "/home/user/projects/my_project";

std::string relative_path = get_relative_path(filename, base_path);

// Output: "src/main.cpp"
```

```cpp
std::string filename = "/var/www/html/index.html";
std::string base_path = "/var/www/xpto";

std::string relative_path = get_relative_path(filename, base_path);

// Output: "../html/index.html"
```

### Testing

Comprehensive tests have been added to ensure the correctness of the get_relative_path function. The tests cover various scenarios, including:

- Standard use cases with different base and target paths.
- Edge cases where the paths are identical.
- Cases with paths involving different directory levels.

Related to https://github.com/aseprite/aseprite/pull/4389

I agree that my contributions are licensed under the MIT License.
You can find a copy of this license at https://opensource.org/licenses/MIT

### CODE REVIEW

File: base/fs.cpp [hunk 1]
```diff
@@ -210,6 +210,47 @@ std::string get_file_title_with_path(const std::string& filename)
   return filename;
 }
 
+std::string get_relative_path(const std::string& filename, const std::string& base_path)
+{
+  std::vector<std::string> baseDirs;
```
Comment (dacap) [inline base/fs.cpp:215]: There is a trailing white space here:

image

File: base/fs.cpp [hunk 2]
```diff
@@ -210,6 +210,47 @@ std::string get_file_title_with_path(const std::string& filename)
...
+    return filename;
+  }
+
+  // Calculate the number of directories to go up from base path
+  std::string relativePath;
+  for (auto it = itFrom; it != baseDirs.end(); ++it) {
+      std::string upDir("..");
+      upDir.push_back(base::path_separator);
+      relativePath += upDir;
+  }
```
Comment (dacap) [inline base/fs.cpp:241]: Probably you can simplify this with
```suggestion
  std::string relativePath;
  for (auto it = itFrom; it != baseDirs.end(); ++it)
    relativePath = base::join_path(relativePath, "..");
```
In other case you should move upDir outside the for-loop, and fix the indentation to 2 spaces.

File: base/fs.cpp [hunk 3]
```diff
@@ -210,6 +210,47 @@ std::string get_file_title_with_path(const std::string& filename)
...
+
+  // Append the remaining part of 'toPath'
+  for (auto it = itTo; it != toParts.end(); ++it) {
+      relativePath += *it;
+      if (it + 1 != toParts.end()) {
+          relativePath.push_back(base::path_separator);
+      }
+  }
```
Comment (dacap) [inline base/fs.cpp:249]: You can use join_path here too:
```suggestion
  for (auto it = itTo; it != toParts.end(); ++it)
    relativePath = base::join_path(relativePath, *it);
```
Or fix the indentation in other case.

**Review by dacap — CHANGES_REQUESTED** [review]
Thanks @gui-marc for this, just some minor changes requested.

**Comment by dacap** [comment]
Thanks for the PR and the fixes @gui-marc :+1:
